lib/saluki-io/src/net/client/http.rs

```diff
@@ -52,4 +53,8 @@ where
     pub async fn send(&self, req: Request<B>) -> Result<Response<Incoming>, Error> {
         self.inner.request(req).await
     }
+
+    pub fn call(&mut self, req: Request<B>) -> ResponseFuture {
+        self.inner.call(req)
+    }
```

Comment:
I think the simpler thing to do here would just be to actually implement `Service<Request<B>>` for `HttpClient<C, B>` rather than create the new `DatadogMetricsService<O>`.